src/Handler/Table/Import/ImportTableFromTableHandler.php

```diff
@@ -222,21 +230,25 @@ private function import(
 
         /** @var TableImportFromTableCommand\SourceTableMapping\ColumnMapping[] $mappings */
         $mappings = iterator_to_array($sourceMapping->getColumnMappings()->getIterator());
-        // if column names are identical, we can use CREATE TABLE COPY (CopyImportFromTableToTable)
-        // if column names are not identical, we have to use INSERT INTO (ToStageImporter)
-        $isColumnNameIdentical = true;
-        foreach ($mappings as $columnMapping) {
-            if ($columnMapping->getSourceColumnName() === $columnMapping->getDestinationColumnName()) {
-                continue;
-            }
-            $isColumnNameIdentical = false;
-        }
         // load to staging table
         $stagingTable = StageTableDefinitionFactory::createStagingTableDefinitionWithMapping(
             $destinationDefinition,
             $mappings,
         );
-        if ($isColumnNameIdentical) {
+
+        $isColumnIdentical = true;
+        try {
+            Assert::assertSameColumns(
+                $sourceTableDefinition->getColumnsDefinitions(),
+                $stagingTable->getColumnsDefinitions(),
+                [ToStageImporterInterface::TIMESTAMP_COLUMN_NAME],
+            );
+        } catch (ColumnsMismatchException) {
+            $isColumnIdentical = false;
+        }
```

Comment:
Tady to je upravená důležitá část. Porovnávám stage protože ta sa dělá z destination bez _timestamp sloupce. Abych mohl použit COPY importer potřebuju mět ty tabulky stejné jinak tam padá cast v BQ, takže sem použil assertSameColumns, který tady to dělá.

Prakticky E2E to pokrývá case kdy je tabulka ve WS s typama a tabulka ve Storage bez typů. Tehdy sa vysere ten cast. Pořád je to validní usecase pro platformu, každopádně už není až tak důležitý protože runner by měl zakládat tabulky z WS s typama podle native-types, takže by tady ten assert neměl padat.

Reply:
Jo vypadá to cajk a dává to smysl.